src/services/web3/web3.service.ts

```diff
@@ -42,6 +35,11 @@ export default class Web3Service {
 
   public setUserProvider(provider: ComputedRef<Web3Provider>) {
     this.userProvider = provider;
+    this.setTxBuilder(provider.value.getSigner());
+  }
+
+  public setTxBuilder(signer: JsonRpcSigner) {
```

Comment:
I don't like the name of this function, as it's inconsistent with setUserProvider above, I would assume the parameter is the TransactionBuilder.